build.go

```diff
@@ -100,40 +102,144 @@ func (m buildModule) getBuildHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, resBuild)
 }
 
-func (m buildModule) getBuild(buildID uint) (database.Build, error) {
-	var dbBuild database.Build
-	if err := m.Database.
-		Where(&database.Build{BuildID: buildID}).
-		Preload(database.BuildFields.TestResultSummaries).
-		Preload(database.BuildFields.Params).
-		First(&dbBuild).
-		Error; err != nil {
-		return database.Build{}, err
-	}
-	return dbBuild, nil
+var buildJSONToColumns = map[string]string{
+	response.BuildJSONFields.BuildID:     database.BuildColumns.BuildID,
+	response.BuildJSONFields.Environment: database.BuildColumns.Environment,
+	response.BuildJSONFields.CompletedOn: database.BuildColumns.CompletedOn,
+	response.BuildJSONFields.ScheduledOn: database.BuildColumns.ScheduledOn,
+	response.BuildJSONFields.StartedOn:   database.BuildColumns.StartedOn,
+	response.BuildJSONFields.Stage:       database.BuildColumns.Stage,
+	response.BuildJSONFields.StatusID:    database.BuildColumns.StatusID,
+	response.BuildJSONFields.IsInvalid:   database.BuildColumns.IsInvalid,
 }
 
-func (m buildModule) getLogs(buildID uint) ([]database.Log, error) {
-	var dbLogs []database.Log
-	if err := m.Database.
-		Where(&database.Build{BuildID: buildID}).
-		Find(&dbLogs).
-		Error; err != nil {
-		return []database.Log{}, err
-	}
-	return dbLogs, nil
-}
+var defaultGetBuildsOrderBy = orderby.Column{Name: database.BuildColumns.BuildID, Direction: orderby.Desc}
 
-// searchBuildListHandler godoc
-// @id searchBuildList
-// @summary NOT IMPLEMENTED YET
+// getBuildListHandler godoc
+// @id getBuildList
+// @summary Get slice of builds.
+// @description List all builds, or a window of builds using the `limit` and `offset` query parameters. Allows optional filtering parameters.
+// @description Verbatim filters will match on the entire string used to find exact matches,
+// @description while the matching filters are meant for searches by humans where it tries to find soft matches and is therefore inaccurate by nature.
 // @tags build
-// @accept json
-// @produce json
-// @success 501 "Not Implemented"
-// @router /builds/search [post]
-func (m buildModule) searchBuildListHandler(c *gin.Context) {
-	c.Status(http.StatusNotImplemented)
+// @param limit query int false "Number of results to return. No limit if unset or non-positive. Required if `offset` is used." default(100)
```

Comment:
Dupe of previous.

Description here doesn't seem to match the changelog description:
> By default all these new endpoints use a default limit of 100, but this can
  be disabled by specifying `?limit=0`.

Assuming the changelog description is the actual behavior?
```suggestion
// @param limit query int false "Number of results to return. Set to 0 or non-positive to disable. Required if `offset` is used." default(100)
```
Alt.
```suggestion
// @param limit query int false "Number of results to return. Set to 0 to disable. Required if `offset` is used." minimum(0) default(100)
```

Reply:
nit: Could you write "Same for the oither `@limit` query params in the other endpoints as well" instead of duplicating the comment? I find it annoying to deal with 5 discussion threads regarding the same thing

I'll mark all but the first thread as resolved